pkg/extensions/sync/remote.go

```diff
@@ -5,192 +5,381 @@ package sync
-			Err(err).Msg("couldn't parse repository reference")
+		return []string{}, err
+	}
 
-		return nil, err
+	tl, err := registry.client.TagList(ctx, repoReference)
+	if err != nil {
+		return []string{}, err
 	}
 
-	var namedRepoRef dockerReference.Named
+	return tl.GetTags()
+}
 
-	digest, ok := parseReference(reference)
-	if ok {
-		namedRepoRef, err = dockerReference.WithDigest(repoRef, digest)
+func convertDockerListToOCI(ctx context.Context, man manifest.Manifest, imageReference ref.Ref, seen []godigest.Digest,
+	regclient *regclient.RegClient,
+) (
+	[]byte, ispec.Descriptor, error,
+) {
+	var index ispec.Index
+
+	// seen
+	if common.Contains(seen, man.GetDescriptor().Digest) {
+		return nil, ispec.Descriptor{}, nil
+	}
+
+	index.SchemaVersion = 2
+	index.Manifests = []ispec.Descriptor{}
+	index.MediaType = ispec.MediaTypeImageIndex
+
+	indexer, ok := man.(manifest.Indexer)
+	if !ok {
+		return nil, ispec.Descriptor{}, zerr.ErrMediaTypeNotSupported
+	}
+
+	ociIndex, err := manifest.OCIIndexFromAny(man.GetOrig())
+	if err != nil {
+		return nil, ispec.Descriptor{}, zerr.ErrMediaTypeNotSupported
+	}
+
+	manifests, err := indexer.GetManifestList()
+	if err != nil {
+		return nil, ispec.Descriptor{}, zerr.ErrMediaTypeNotSupported
+	}
+
+	for _, manDesc := range manifests {
+		ref := imageReference
+		ref.Digest = manDesc.Digest.String()
+
+		manEntry, err := regclient.ManifestGet(ctx, ref)
 		if err != nil {
-			return nil, err
+			return nil, ispec.Descriptor{}, err
 		}
-	} else {
-		namedRepoRef, err = dockerReference.WithTag(repoRef, reference)
-		if err != nil {
-			return nil, err
+
+		regclient.Close(ctx, manEntry.GetRef())
+
+		var desc ispec.Descriptor
+
+		switch manEntry.GetDescriptor().MediaType {
+		case mediatype.Docker2Manifest:
+			_, desc, err = convertDockerManifestToOCI(ctx, manEntry, ref, regclient)
+			if err != nil {
+				return nil, ispec.Descriptor{}, err
+			}
+
+		case mediatype.Docker2ManifestList:
+			_, desc, err = convertDockerListToOCI(ctx, manEntry, ref, seen, regclient)
+			if err != nil {
+				return nil, ispec.Descriptor{}, err
+			}
+		default:
+			return nil, ispec.Descriptor{}, err
+		}
+
+		// copy desc platform from docker desc
+		if manDesc.Platform != nil {
+			desc.Platform = &ispec.Platform{
+				Architecture: manDesc.Platform.Architecture,
+				OS:           manDesc.Platform.OS,
+				OSVersion:    manDesc.Platform.OSVersion,
+				OSFeatures:   manDesc.Platform.OSFeatures,
+				Variant:      manDesc.Platform.Variant,
+			}
 		}
+
+		index.Manifests = append(index.Manifests, desc)
 	}
 
-	imageRef, err := docker.NewReference(namedRepoRef)
-	if err != nil {
-		registry.log.Err(err).Str("transport", docker.Transport.Name()).Str("reference", namedRepoRef.String()).
-			Msg("cannot obtain a valid image reference for given transport and reference")
+	index.Annotations = ociIndex.Annotations
+	index.ArtifactType = ociIndex.ArtifactType
 
-		return nil, err
+	if ociIndex.Subject != nil {
```

Comment:
Can we get better coverage of this?

Reply:
actually I removed it, docker image can not have subject.